onyo/lib/tests/test_onyo.py

```diff
@@ -0,0 +1,50 @@
+from pathlib import Path
+
+import pytest
+from onyo import OnyoRepo, OnyoInvalidRepoError
+
+
+def test_OnyoRepo_instantiation_existing(repo: OnyoRepo) -> None:
+    """
+    The OnyoRepo class must instantiate correctly for paths to existing
+    repositories.
+    """
+    new_repo = OnyoRepo(repo.git.root, init=False)
+    assert new_repo.git.root.samefile(repo.git.root)
+
+
+def test_OnyoRepo_instantiation_non_existing(tmp_path: Path) -> None:
+    """
+    The OnyoRepo class must instantiate correctly for paths to non-existing
+    repositories.
+    """
+    new_repo = OnyoRepo(tmp_path, init=True)
+    assert new_repo.git.root.samefile(tmp_path)
+    assert (new_repo.git.root / ".onyo").exists()
```

Comment:
I'd add an assertion on a clean repo /fsck here as well, since so far this only tests what's written to disc - not what's committed.